tools/Dockerfile

```diff
@@ -36,10 +39,12 @@ RUN set -eux; \
 #
 FROM bare_base_image AS install_rust
 ARG rust_stable_version=1.56.1
-ARG rust_nightly_version=nightly-2022-03-03
+ARG rust_nightly_version=nightly-2022-03-29
 ARG cargo_udeps_version=0.1.27
 ARG cargo_hack_version=0.5.12
-ARG smithy_rs_revision=main
+# If the `checkout_smithy_rs_tools` arg is set to true, then the Dockerfile will acquire the tools
+# source code by checking out awslabs/smithy-rs/main rather than copying them from the local directory.
+ARG checkout_smithy_rs_tools=false
```

Comment:
do we even need this arg 🤔 

We could have the other docker build handle the repo clone separately from building...